File: samples/src/commonMain/kotlin/sample/_class/cascade/Cascade.ts
import { runTest } from "../../shared_ts/RunTest"
import { assert } from "../../shared_ts/Assert"
import Samples from '@kustom/Samples'

runTest("Cascade", () : void => {
    var parking = new Samples.sample._class.cascade.js.Parking()
    //var parking = new Samples.Parking() // erasePackage=true
    assert(parking.currentCars.length == 2, "direct child")
    assert(parking.currentCars[1].wheels[3].diameter == 22, "sub childs")
})

File: samples/src/commonMain/kotlin/sample/_class/cascade/Cascade.kt
package sample._class.cascade

import deezer.kustomexport.KustomExport

@KustomExport
class Wheel {
    val diameter: Long = 22
}

@KustomExport
class Car {
    val wheels = listOf<Wheel>(Wheel(), Wheel(), Wheel(), Wheel())
}

@KustomExport
class Parking {
    val currentCars = listOf(Car(), Car())
}
